examples/h3server.rs

```diff
@@ -0,0 +1,396 @@
+// Copyright (C) 2019, Cloudflare, Inc.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+// IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#[macro_use]
+extern crate log;
+
+use std::net;
+
+use std::collections::HashMap;
+
+use ring::rand::*;
+
+const LOCAL_CONN_ID_LEN: usize = 16;
+
+const MAX_DATAGRAM_SIZE: usize = 1452;
+
+const USAGE: &str = "Usage:
+  h3server [options]
+  h3server -h | --help
+
+Options:
+  --listen <addr>   Listen on the given IP:port [default: 127.0.0.1:4433]
+  --cert <file>     TLS certificate path [default: examples/cert.crt]
+  --key <file>      TLS certificate key path [default: examples/cert.key]
+  --root <dir>      Root directory [default: examples/root/]
+  --name <str>      Name of the server [default: quic.tech]
+  -h --help         Show this screen.
+";
+type QuicH3Conn = (Box<quiche::Connection>, Option<quiche::h3::Connection>);
```

Comment:
Nit: need empty line before this.

Reply:
Also, it's probably better to make this a proper `struct` and name the fields. Reading the code where they are used is pretty confusing.